cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/StatefulSetDiff.java

```diff
@@ -21,7 +21,7 @@
 
     private static final Pattern IGNORABLE_PATHS = Pattern.compile(
         "^(/spec/revisionHistoryLimit"
-        + "|/spec/template/metadata/annotations"  // Actually it's only the statefulset-generation annotation we care about
+        + "|/spec/template/metadata/annotations/strimzi.io~1generation"
```

Comment:
Presumably the `~1` is some kind of encoding for the `/` that can't be included because it would be interpreted as a component of the path.